src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/CaseNoteFilter.kt

```diff
@@ -0,0 +1,84 @@
+package uk.gov.justice.hmpps.prison.repository.jpa.repository
+
+import com.google.common.collect.ImmutableList
+import jakarta.persistence.criteria.CriteriaBuilder
+import jakarta.persistence.criteria.CriteriaQuery
+import jakarta.persistence.criteria.Predicate
+import jakarta.persistence.criteria.Root
+import org.springframework.data.jpa.domain.Specification
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderCaseNote
+import uk.gov.justice.hmpps.prison.service.BadRequestException
+import uk.gov.justice.hmpps.prison.service.QueryParamHelper
+import java.time.LocalDate
+
+class CaseNoteFilter(
+  val bookingId: Long,
```

Comment:
I've made the first field here not nullable as `bookingService.getLatestBookingByOffenderNo` throws an exception if the offender isn't found and hence the booking